spec-0014/index.md

```diff
@@ -0,0 +1,410 @@
+
+- API reference documentation, and
+- narrative long-form content
+
+For both of these, we recommend different approaches to enable interactivity:
+
+###### API documentation
+
+`jupyterlite-sphinx` offers the [`try_examples` directive](https://jupyterlite-sphinx.readthedocs.io/en/stable/directives/try_examples.html), which can make "Examples" sections within API documentation interactive. This directive adds buttons which swap the rendered code snippets in-place with an interactive mini-notebook that can be executed without leaving the page. There is an option to have the directives inserted into "Examples" sections automatically for documentation that uses [numpydoc](https://numpydoc.readthedocs.io/), and with [`sphinx.ext.napoleon`](https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html) supported on a best-effort basis. The look, shape, feel, text, and inclusion of these buttons across the documentation can be configured through user-defined options.
+
+###### Narrative documentation and tutorials (long-form content)
+
+For longer-form narrative content such as tutorials, how-to guides, and explanatory documentation:
+
+- `jupyterlite-sphinx` offers directives to render long-form content written in notebooks (the [`jupyterlite` directive](https://jupyterlite-sphinx.readthedocs.io/en/stable/directives/jupyterlite.html), the [`notebooklite` directive](https://jupyterlite-sphinx.readthedocs.io/en/stable/directives/notebooklite.html), and so on), which can be configured to open notebooks in new tabs or iframes within the same tab. It supports both traditional IPyNB files and MyST Markdown notebooks. Unlike the `try_examples` directive, these directives do not work with reST-based content. We recommend using MyST Markdown notebooks, as they integrate well with version control.
+- The Sphinx-Gallery project offers a [JupyterLite integration, built with `jupyterlite-sphinx`](https://sphinx-gallery.github.io/stable/configuration.html#jupyterlite) which enables a "JupyterLite" button in the secondary sidebar to open a notebook in the JupyterLite deployments. It provides a `notebook_modification_function` configuration option via `conf.py` that can be used to configure code cells or Markdown cells at the top of a notebook, with custom `%pip`-install commands, imports from other libraries, or extra admonitions.
+
+---
+
+Once interactive documentation has been configured and deployed, the ["Updating package and kernel versions for Pyodide-powered or emscripten-forge-powered websites"](#Updating-package-and-kernel-versions-for-Pyodide-powered-or-emscripten-forge-powered-websites) section contains information on how to increment the packages' and distribution/kernel's versions.
+
+An end-to-end example of the above steps is also available; see [the `jupyterlite-sphinx-demo` project](#an-end-to-end-example).
+
+#### What distribution should projects choose: Pyodide or emscripten-forge?
```

Comment:
I think this section needs a bit of work. In the end, it's not clear at all to me why I would choose one project over the other.

My recommendation for structure here would be to start with an outline, like:

- Pros/cons of Pyodide
- Pros/cons of emscripten-forge

Then, if you want to have more discussion that can be valuable, but it will be easier to grasp the main differences when reading.

Reply:
I have converted all this information into a table in 271a417f34274a93a00cd611f72d594cd01d6421 which makes things way more concise and streamlined, do you like it?

Reply:
Also, cc: @steppi, as you also previously expressed your concerns about this section above in your reviews; I hope the rewrite makes sense.

Reply:
Looks much clearer, thank you!